### DISCUSSION TITLE
Accessing occasionally from Windows with Desktop

### DISCUSSION THREAD

**Discussion by interstar** [community/community#24065] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I use github typically from Linux using standard git terminal commands to push and pull. I have my github account set up to use ssh authentication.</p>
<p>But today I needed to access the account from a Windows machine. I don’t have this machine specifically configured for this. But I figured if I just installed Github Desktop for Windows it would let me do what I needed.</p>
<p>However I can’t seem to find a way to get this Desktop client to connect with my github account. I’ve theoretically logged in to the account with name and password. But I can’t push the changes I made to a repository</p>
<p>When I try I get :</p>
<pre><code class="lang-auto">

Authentication failed. Some common reasons include:

- You are not logged in to your account: see File &gt; Options.
- You may need to log out and log back in to refresh your token.
- You do not have permission to access this repository.
- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
- If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account.
- If you use SSH authentication, ensure the host key verification passes for your repository hosting service.
- If you used username / password authentication, you might need to use a Personal Access Token instead of your account password. Check the documentation of your repository hosting service.
</code></pre>
<p>Is there no way I can generate an ssh key from inside of and only for the use within github Desktop client, which I can then associate with my account?</p>

**Comment by steveward** [comment, marked as answer]
<p><a class="mention" href="/interstar">@interstar</a> you can use HTTPS with GitHub Desktop, which will work by default if you are logged in. You will need to change the remote URL to use HTTPS:</p>
<p><a href="https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/working-with-your-remote-repository-on-github-or-github-enterprise/changing-a-remotes-url-from-github-desktop" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/working-with-your-remote-repository-on-github-or-github-enterprise/changing-a-remotes-url-from-github-desktop</a></p>
<p>If you want to use SSH you will need to follow the steps to set up a new SSH key on Windows:</p>
<p><a href="https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent</a></p>
<p>I hope that helps!</p>
